src/charm_state.py

```diff
@@ -266,6 +268,35 @@ def userids_to_list(cls, value: str) -> typing.List[str]:
                 raise ValidationError(f"Invalid user ID format: {user_id}", cls)
         return value_list
 
+    @validator("ready_check")
+    @classmethod
+    def to_pebble_check(cls, value: str) -> typing.Dict[str, typing.Union[str, int]]:
+        """Convert the ready_check field to pebble check.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The pebble check.
+
+        Raises:
+            ValidationError: if ready_check is invalid.
+        """
+        # expected
+        # period;threshold;timeout
+        config_values = value.split(",")
```

Comment:
Are the values delimited by semicolons or commas?

Reply:
Commas, fixed.